### PR TITLE
Bump ff & group to 0.13

### LINKED ISSUE #171: Upgrading ff and group crates
This is issue is meant as a reminder and not as an immediate action item.  There are new releases of `ff` and `group`. Upgrading to those is a breaking change as they contain traits and you cannot really have the traits of two different versions in your dependency tree.

I propose postponing the upgrade until a new breaking release is needed for other reasons. This upgrade could then combined with such a release.

The upgrade is not straight forward, as also all other dependencies using those traits would need to be updates e.g. `bellperson`.

The upgrade would enable an upgrade of `pasta_curves` as well. The most recent release v0.5.0 should contain everything our current fork contains. This means once upgraded, we won't need to rely on a fork anymore.

### PR DESCRIPTION
**Pull request by huitseeker**
~~Still testing this, in particular, the rust-gpu-tools import on the cuda / opencl features may have some issues.~~

Edit: solved by upgrading rust-gpu-tools to 0.7.1 in upstream `ec_gpu`.

Depends on https://github.com/filecoin-project/bellperson/pull/304
Fixes #171 

~~Edit(4/14): this needs work on the pasta-fil-curves repo.~~
Edit. (4/15): I'm seeing issues with tree builder tests hanging on opencl. Is this something you're seeing as well @porcuquine ?

**Historical (solved) issue:**
https://gist.github.com/huitseeker/230caa8cd63e049846f95ac0d8c77fe5

```
huitseeker@lightning.local➜~/tmp/neptune(bump_ff_group)» cargo tree -p rust-gpu-tools -i                                                                                                                                             [12:48:10]
warning: patch for `blstrs` uses the features mechanism. default-features and features will not take effect because the patch dependency does not support this mechanism
rust-gpu-tools v0.7.0
└── ec-gpu-gen v0.5.2 (https://github.com/filecoin-project/ec-gpu/?branch=upgrade-ff-013#9616bd3c)
    ├── bellperson v0.24.1 (https://github.com/huitseeker/bellperson?branch=update-ff-group#36a75691)
    │   ├── gbench v0.5.4 (/Users/huitseeker/tmp/neptune/gbench)
    │   └── neptune v8.1.1 (/Users/huitseeker/tmp/neptune)
    │       └── gbench v0.5.4 (/Users/huitseeker/tmp/neptune/gbench)
    ├── gbench v0.5.4 (/Users/huitseeker/tmp/neptune/gbench)
    └── neptune v8.1.1 (/Users/huitseeker/tmp/neptune) (*)
    [build-dependencies]
    ├── bellperson v0.24.1 (https://github.com/huitseeker/bellperson?branch=update-ff-group#36a75691) (*)
    └── neptune v8.1.1 (/Users/huitseeker/tmp/neptune) (*)
    ```

### CODE REVIEW

**Comment by vmx** [comment]
The problem was that `CudaContexts` wasn't `Send`. I've just released version 0.7.1 which should fix the issue.

**Comment by vmx** [comment]
> Edit(4/14): this needs work on the pasta-fil-curves repo.

The nice thing about this upgrade is that `fil_pasta_curves` should no longer be needed. Upstream `pasta_curves` should just work.

**Comment by huitseeker** [comment]
@vmx Correct. A msiconfig in test flags momentarily made me think there was something missing in `ec_gpu`. Fixed in [267bf6d](https://github.com/lurk-lab/neptune/pull/179/commits/267bf6d5d3a09bf684a85ad2c4499f7a9564b97f).

File: gbench/Cargo.toml [hunk 1]
```diff
@@ -7,20 +7,19 @@ edition = "2018"
 license = "MIT OR Apache-2.0"
 
 [dependencies]
-lazy_static = "1.4.0"
 bellperson = { version = "0.24.1", default-features = false }
```
Comment (samuelburnham) [inline gbench/Cargo.toml:10]: Any reason not to update to bellperson v0.25?

**Review by samuelburnham — APPROVED** [review]

**Review by porcuquine — APPROVED** [review]

**Comment by huitseeker** [comment]
Took a while, but I can now confirm this works on both cuda and opencl:
```
EC_GPU_FRAMEWORK=opencl cargo test --release  --features "arity2,arity4,arity8,arity11,arity16,arity24,arity36,pasta,opencl,bls" -- --test-threads=1 --nocapture
```
https://gist.github.com/huitseeker/e79b04794276989604c646d296662d27
```
EC_GPU_FRAMEWORK=cuda cargo test --release  --features "arity2,arity4,arity8,arity11,arity16,arity24,arity36,pasta,cuda,bls" -- --test-threads=1 --nocapture
```
https://gist.github.com/huitseeker/98f65366722d5296bcc4037081e41bb9

**Pull request merged**